src/main/java/org/candlepin/pki/certs/SCACertificateGenerator.java

```diff
@@ -58,216 +65,333 @@
+        List<Environment> environments = environmentCurator.getConsumerEnvironments(consumer);
 
-    private SCACertificate wrap(SCACertificate cert) {
-        SCACertificate result = new SCACertificate();
-        result.setCert(cert.getCert() + cert.getContent());
-        result.setCreated(cert.getCreated());
-        result.setUpdated(cert.getUpdated());
-        result.setId(cert.getId());
-        result.setConsumer(cert.getConsumer());
-        result.setKey(cert.getKey());
-        result.setSerial(cert.getSerial());
-        return result;
+        return getContentAccessPayload(owner, consumer, environments);
     }
 
     /**
-     * Method create a new {@link SCACertificate}. If certificate
-     * already exists it is reused and updated.
+     * Retrieves the X509 certificate component of a {@link SCACertificate} for a provided v3 cert capable
+     * consumer that is part of an owner using simple content access. The X509 certificate will be generated
+     * for the consumer if one does not exist or if the existing certificate is expired. A certificate is
+     * considered expired if the {@link CertificateSerial}'s expiration date is after the configured X509
+     * certificate exiration threshold.
+     *
+     * @param consumer
+     *  the consumer to retrieve a X509 certificate for
+     *
+     * @throws IllegalArgumentException
+     *  if the provided consumer is null
      *
-     * @param consumer A consumer for which to create a certificate
-     * @param owner An owner to be used for certificate creation
-     * @return Content access certificate
+     * @return the X509 certificate for the consumer, or null if the owner is not in SCA mode or the
+     *  consumer is not v3 cert capable
      */
-    private SCACertificate generateFor(Consumer consumer, Owner owner) {
-        SCACertificate existing = consumer.getContentAccessCert();
+    public SCACertificate getX509Certificate(Consumer consumer) {
+
+        if (consumer == null) {
+            throw new IllegalArgumentException("consumer is null");
+        }
+
+        Owner owner = consumer.getOwner();
+        if (owner == null || !owner.isUsingSimpleContentAccess()) {
+            log.debug("Consumer is not in SCA mode");
+            return null;
+        }
 
-        SCACertificate certificate = existing != null ?
-            this.updateCertificate(consumer, owner, existing) :
-            this.createCertificate(consumer, owner);
+        if (!v3CapabilityCheck.isCertV3Capable(consumer)) {
+            log.debug("Consumer is not v3 cert capable");
+            return null;
+        }
 
-        // Due to the shenanigans we do to wrap and juggle the data in this cert, it's critical we invoke
-        // Hibernate's persist and update hooks by flushing the creation/update operations above so the
-        // dates are correct in the output cert object.
-        this.contentAccessCertificateCurator.flush();
```

Comment:
I think we need to preserve this flush, since when we're later doing this: `combined.setUpdated(payloadUpdated.after(certUpdated) ? payloadUpdated : certUpdated);` we could get in a situation were we generated a brand new X509 cert that has a null `certUpdated` date, so we end up setting the date to `payloadUpdated` here, which could be quite old. So we would end up sending back a very old date to the client, which it would send back as an old if-modified-since date, resulting in churn for the getContentAccessBody endpoint

Reply:
We don't. We're already getting flushes on each create op, and even those aren't strictly necessary for all paths.